File: Python/d01.py
with open("data/d01.txt") as f:
    input = f.read().split("\n")

def solve(input:list[str], top:int) -> int:
    '''
    Solves day 1 of AOC 2022
    Parameters: input -- list[str], top -- int
    Returns: value -- int
    '''
    input = [int(x) if x not in '' else None for x in input]
    splits = [i for i, e in enumerate(input) if e == None]
    splits.append(len(input))
    res = []
    start = 0

    for ii in range(len(splits)):
        res.append(sum(input[start:splits[ii]]))
        start = splits[ii] + 1  

    res.sort(reverse=True)

    return sum(res[0:top])

# Part 1 Answer
print(solve(input, 1))

# Part 2 Answer
print(solve(input, 3))

File: R/d01.r
#' Solve day 1 AOC 2022
#'
#' @param input puzzle input
#' @param top integer specifying how many elements to include
#'  in the returned sum.
#' @return a numeric; the sum of the top n results
#' @export
solve = function(input, top) {
  input = ._process(input)
  return(sum(sort(unlist(input), decreasing = TRUE)[1:top]))
}

._process = function(data) {
  data = as.double(data)
  splits = c(which(is.na(data)), length(data))
  res = vector(mode = "list", length = length(splits))
  start = 1

  for (ii in seq_len(length(splits))) {
    res[ii] = sum(data[start:(splits[ii] - 1)])
    start = splits[ii] + 1
  }

  return(res)
}
